src/mlpack/tests/spsa_test.cpp

```diff
@@ -0,0 +1,50 @@
+/**
+ * @file spsa_test.cpp
+ * @author N Rajiv Vaidyanathan
+ *
+ * Test file for SPSA (simultaneous pertubation stochastic approximation).
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/core/optimizers/spsa/spsa.hpp>
+#include <mlpack/core/optimizers/problems/sgd_test_function.hpp>
+#include <mlpack/core/optimizers/problems/generalized_rosenbrock_function.hpp>
+#include <mlpack/core/optimizers/problems/sphere_function.hpp>
+#include <mlpack/methods/logistic_regression/logistic_regression_function.hpp>
```

Comment:
Since you include the logistic regression function here, do you think we could add a test for this function as well?

Reply:
Yeah sure!